src/app/util/af-types.h

```diff
@@ -236,6 +236,13 @@ struct EmberAfDefinedEndpoint
      * Span pointing to a list of tags. Lifetime has to outlive usage, and data is owned by callers.
      */
     chip::Span<const chip::app::Clusters::Descriptor::Structs::SemanticTagStruct::Type> tagList;
+
+    /**
+     * Unique Id for this end point.
+     */
+    char endpointUniqueId[chip::app::Clusters::Descriptor::Attributes::EndpointUniqueID::TypeInfo::MaxLength()] = { 0 };
```

Comment:
So as you might have noticed... this has RAM costs, since we are allocating this storage unconditionally, even for applications that don't want to implement the new optional attribute.  Should this be conditioned on some build flag?

Alternately, you could use a setup similar to TagList and have minimal storage costs here (basically two words) and the rest of the cost is paid by the application, if it decides to provide this data.